src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
             }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
         }
 
-        // check if a class is noted as abstract in the text
-        return psiClass.text.replace(" ", "")
-            .contains("abstractclass${psiClass.name}", ignoreCase = true)
+    override val superClass: PsiClassWrapper? get() = psiClass.superClass?.let { JavaPsiClassWrapper(it) }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override fun searchSubclasses(project: Project): Collection<PsiClassWrapper> {
+        val scope = GlobalSearchScope.projectScope(project)
+        val query = ClassInheritorsSearch.search(psiClass, scope, false)
+        return query.findAll().map { JavaPsiClassWrapper(it) }
     }
 
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingMethods = mutableSetOf(psiMethod as JavaPsiMethodWrapper)
+        for (currentPsiMethod in allMethods) {
+            if ((currentPsiMethod as JavaPsiMethodWrapper).isConstructor) interestingMethods.add(currentPsiMethod)
+        }
+        val interestingPsiClasses = mutableSetOf(this)
+        interestingMethods.forEach { methodIt ->
+            methodIt.parameterList.parameters.forEach { paramIt ->
+                PsiTypesUtil.getPsiClass(paramIt.type)?.let { typeIt ->
+                    JavaPsiClassWrapper(typeIt).let {
+                        if (!it.qualifiedName.startsWith("java.")) {
+                            interestingPsiClasses.add(it)
+                        }
+                    }
+                }
+            }
+        }
+
+        return interestingPsiClasses.toMutableSet()
+    }
+
+    val isInterface: Boolean get() = psiClass.isInterface
+
+    val isAbstractClass: Boolean
+        get() {
+            psiClass.containingFile.virtualFile
+            if (psiClass.isInterface) return false
+
+            val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
+            for (psiMethod: PsiMethod in methods) {
+                if (psiMethod.body == null) {
+                    return true
+                }
+            }
+
+            // check if a class is noted as abstract in the text
+            return psiClass.text.replace(" ", "")
+                .contains("abstractclass${psiClass.name}", ignoreCase = true)
+        }
```

Comment:
Arkadii has the exact implementation for a similar method being introduced in this [PR](https://github.com/JetBrains-Research/TestSpark/pull/226/files/8080207ecf6cbe71b0213de6dc68fa9525099c72#diff-ab82c750e87a21a7b3d95c4b1f189be75e7ee96f626da05fca16b39823425507) (`ClassTypeProcessHelper.isAbstract`). We will have a duplicate here, which is bad; we need to avoid it somehow.

@arksap2002

Reply:
I canceled that PR due to incorrect implementation in case of language increasing

Reply:
pushed a new changes in this branch